nengo_spa/algebras/base.py

```diff
@@ -220,6 +301,66 @@ def implement_binding(self, n_neurons_per_d, d, unbind_left, unbind_right):
         """
         raise NotImplementedError()
 
+    def sign(self, v):
+        """Returns the sign of *v* defined by the algebra.
+
+        The exact definition of the sign depends on the concrete algebra, but
+        should be analogous to the sign of a number in so far that
+
+        * binding of two "positive" vectors produces another "positive" vector,
+        * binding of a "negative" vector with itself produces a "positive"
+          vector,
+        * binding of a "negative" and "positive" vectors produces a "negative"
+          vector.
```

Comment:
After writing https://github.com/nengo/nengo-spa/pull/271/files#r595699304 I realized this might mislead people as one might expect that binding one "negative" vector with another "negatve" vector would be guaranteed to produce a "positive" vector. However, in the HRR algebra, with even dimensionality, there are three ways that a vector can be "negative" based on the combination of the two coefficients: `A = (-1, 1)`, `B = (1, -1)`, and `C = (-1, -1)`. Then we have:

 `*` | A | B | C |
 --- | --- | --- | ---
 A | `+` | `-` | `-`
 B | `-` | `+` | `-` 
 C | `-` | `-` | `+` 

so the analogy breaks if you are not in the same sub-group for the "negative" times "negative" case. The docstring says "with itself" so that is okay but it might still defy people's expectations.

Complex numbers solve this by not reducing the sign to "positive" or "negative" but keeping it as `z / abs(z)` (keeping phase and throwing away amplitude). Just as we are keeping it as this two-tuple and throwing away all the other coefficients. Then `sign(X*Y) = sign(X)*sign(Y)` is the sort of thing that I think a user can expect to hold in general, as it does here.

So I guess my suggestion then is to say here that it is analogous because of `sign(X*Y) = sign(X)*sign(Y)` and then for the HRR sign, you can define what it means to mulyiply `sign(X)` and `sign(Y)` by multiplying those tuples element-wise with one another.

Reply:
* [ ] This is a good point that should be added do the documentation.

I will also have a look whether some general documentation/user guide besides the bare API documentation should be added.